### ISSUE TITLE
sending a nodejs buffer

### ISSUE THREAD

**Issue by CraigglesO** [node-webrtc/node-webrtc#303]
RTCDataChannelMessageEvent { data: ArrayBuffer { byteLength: 8192 } }

TypeError: Invalid data, chunk must be a string or buffer, not object

The problem is even if I set channel.binaryType = 'blob'; I still get this error If I send a buffer.

And a buffer `new Buffer(5)`
could return above (byteLength of 8192)
Which is absurd.. I send 5 bytes not 8192.

What is the recommended method of sending buffers?

Thanks.

edit - wouldn't just converting it to an arraybuffer before sending it off be a big slow down on performance?

**Comment by CraigglesO** [comment]
Update:

```
send(payload: ArrayBuffer | Buffer | string | Blob | Object) {
    if (Buffer.isBuffer(payload))
      payload = new Uint8Array(payload);
    this.dataChannels['reliable'].send(payload);
  }
```


```
channel.onmessage = function(event) {
        let data;
        if (event.data instanceof ArrayBuffer) data = new Buffer(event.data)
        else data = event.data;
        self.emit('data', data);
      };
```

This works...
Is it a lazy solution?

**Comment by webarchymeta** [comment]
:-) I was having this problem as well and before I turned around to find a solution, a base64 encoding of data was applied temporarily. If you do not, I am going to raise similar question. Thanks you for your solution! 

In fact it seemed that I tried your conversions, but something wasn't done right at the time ... but never mind ...

**Comment by markandrus** [comment]
This was actually fixed in 0.0.65 :-)
